### PR TITLE
docs: clarify tuning iterations and population size

### LINKED ISSUE #24279: model.tune(): How is population size defined and how can it be controlled?
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

## Description
I am using Ultralytics YOLO's built-in hyperparameter tuning via `model.tune()` (without Ray Tune), and I would like to better understand how the genetic algorithm is configured.

## Question
How is the **population size** defined in the default genetic algorithm?

- Is it fixed internally?
- Is it derived from the number of iterations?
- Is there a way to explicitly control or override it?

## What I tried
I ran:
~~~python
from ultralytics import YOLO

model = YOLO("yolo12n.pt")
model.tune(data="data.yaml", epochs=50, iterations=40, use_ray=False)
~~~

## Environment
OS: Linux
Python: 3.11.15
Ultralytics: 8.3.63
Torch: 2.7.0+cu128
GPU: NVIDIA RTX 5090

### Additional

The tuning process runs successfully, but I want to better understand how the search space is explored and whether adjusting population size could improve results.

### PR DESCRIPTION
**Pull request by xiazl1993**
## Summary
- clarify that built-in model.tune(use_ray=False) runs one hyperparameter trial per iteration
- explain that there is no separate population size argument in the built-in genetic tuner
- point users to Ray Tune for parallel trials and advanced search algorithms

Closes #24279

## Tests
- git diff --check
- python3 basic Markdown structure check

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Clarifies how hyperparameter tuning `iterations` works in the built-in tuner and explains that there is no separate population size parameter. 📚

### 📊 Key Changes
- Adds a new **Iterations and Population Size** subsection to `docs/en/guides/hyperparameter-tuning.md`.
- Explains that with `use_ray=False`, `iterations` means the total number of sequential tuning trials, where each iteration trains one model with one hyperparameter configuration.
- Provides a concrete example showing that `iterations=40` and `epochs=50` can schedule up to 40 separate 50-epoch training runs.
- Documents that the built-in genetic tuner does not expose a separate population size argument.
- Clarifies that the tuner selects from up to nine top-fitness prior configurations as a parent pool, then generates one new candidate per iteration.
- Notes that `use_ray=True` enables Ray Tune behavior, where `iterations` is passed as `num_samples` for parallel or more advanced search workflows.

### 🎯 Purpose & Impact
- Reduces confusion for users interpreting `iterations` as a population size or as a single training run setting. ✅
- Improves documentation accuracy around the built-in genetic tuning workflow.
- Helps users better estimate training time and compute requirements before launching tuning jobs. ⏱️
- Guides advanced users toward Ray Tune when they need parallel trials or more sophisticated search strategies. 🚀

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @xiazl1993, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness, and an engineer will assist you soon. Please review the checklist below to help ensure a smooth integration of your documentation update 📚✨

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Verified the doc changes against the current tuning implementation and didn't find any genuine issues in the added section. The clarification is accurate and helpful for distinguishing native sequential tuning from Ray Tune behavior.

**Comment by xiazl1993** [comment]
I have read the CLA Document and I sign the CLA

**Comment by xiazl1993** [comment]
recheck

**Comment by xiazl1993** [comment]
I have read the CLA Document and I sign the CLA

**Comment by xiazl1993** [comment]
recheck

**Comment by glenn-jocher** [comment]
Thanks, the CLA is recorded now, so no further action is needed from your side and the PR can proceed to maintainer review.

**Comment by Laughing-q** [comment]
@xiazl1993 merging! thank you for your contribution!
